validator/validator/metrics.py

```diff
@@ -42,3 +43,15 @@ def calculate_score(self, uid: int) -> float:
             0.0,
             self.baseline_averages[uid] - self.model_averages[uid]
         ) * self.similarity_averages[uid]
+
+    def __getstate__(self):
+        return {
+            "baseline_averages": self.baseline_averages,
+            "model_averages": self.model_averages,
+            "similarity_averages": self.similarity_averages,
+        }
+
+    def __setstate__(self, state):
+        self.baseline_averages = state.get("baseline_averages", [0.0] * self.metagraph.n.item())
+        self.model_averages = state.get("model_averages", [0.0] * self.metagraph.n.item())
+        self.similarity_averages = state.get("similarity_averages", [0.0] * self.metagraph.n.item())
```

Comment:
You can use `self.__dict__` to get the members as a dict(and then exclude `metagraph`, and then to set state you can use `self.__dict__.update(state)`.

Reply:
This might also have a bug, in that metagraph is not set upon loading, not sure how that's addressed though.